docs/src/main/asciidoc/reaugmentation.adoc

```diff
@@ -0,0 +1,67 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Re-augment a Quarkus Application
+
+include::./attributes.adoc[]
+
+== What is augmentation?
+
+Quarkus application configuration may include two types of configuration options: build time options, that are handled during the application build time, and runtime options that may be adjusted after the application has been built but before it has been launched.
+The augmentation is a phase of an application build process during which the byte code of the application is optimized according to the application build time configuration.
+Initialization steps that used to happen when an EAR file was deployed on a Jakarta-EE server such as parsing static configuration, creating proxy instances, etc. now happen at augmentation time.
+CDI beans added after augmentation won't work (because of the missing proxy classes) as well as build-time properties (e.g. `quarkus.datasource.db-kind`) changed after augmentation will be ignored.
+Build-time properties are marked with a lock icon (icon:lock[]) in the link:all-config[list of all configuration options].
+It doesn't matter if you use profiles or any other way to override the properties.
+The build-time properties that were active during augmentation are baked into the build.
+
+> Re-augmentation is the process of recreating the augmentation output for a different build time configuration
+
+== When is re-augmentation useful?
+
+Re-augmentation is useful in case the users of your application want to be able to change some of its build-time properties.
+For instance changing the database driver or switching features on or off (e.g. jaeger).
+If there are only two or three build-time properties that depend on the user environment, you may consider providing alternative builds of the application. However, in case there are more such properties you may prefer shipping a mutable jar instead and let your users re-augment the application for their environment.
```

Comment:
```suggestion
If there are only two or three build time properties that depend on the user environment, you may consider providing alternative builds of the application. However, in case there are more such properties you may prefer shipping a mutable jar instead and let your users re-augment the application for their environment.
```